src/ImageBuilder.tsx

```diff
@@ -20,10 +20,14 @@ async function buildImage(
     "/services/binder/build/",
     window.location.origin,
   );
+  const headers: Record<string, string> = {};
+  if (window.jupyter?.token && window.jupyter.token.length > 0) {
+    headers.Authorization = `token ${window.jupyter.token}`;
+  }
```

Comment:
You can use the native `Headers` implementation for full type safety, yada yada...

```suggestion
  const headers = new Headers();
  if (window.jupyter?.token && window.jupyter.token.length > 0) {
    headers.append("Authorizaiton", `token ${window.jupyter.token}`);
  }
```